rpc-broker/src/rpc-broker.c

```diff
@@ -0,0 +1,444 @@
+/*
+ Copyright (c) 2018 AIS, Inc.
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+*/
+
+#define _GNU_SOURCE
+#include <errno.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <signal.h>
+#include <sys/select.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <uv.h>
+
+#include "rpc-broker.h"
+
+/*
+ * Whenever clients connect to port 5555, this function will then connect
+ * directly to the DBus system bus socket (/var/run/dbus/system_bus_socket).
+ *
+ * Polling on the client & server file-descriptors until the connection
+ * communication is finished.
+ */
+void broker_message(struct raw_dbus_conn *rdconn)
+{
+    int srv = rdconn->server;
+    int domid = rdconn->client_domain;
+    int client = rdconn->client;
+    int sret = 1, cret = 1;
+
+    while (sret > 0 || cret > 0) {
+        cret = exchange(client, srv, domid, recv, send);
+        if (cret < 0)
+            break;
+        sret = exchange(srv, client, domid, recv, send);
+    }
+}
+
+signed int is_stubdom(uint16_t domid)
+{
+    size_t len = 0;
+#ifdef HAVE_XENSTORE
```

Comment:
Would this ever be ran in an environment without XenStore? Perhaps use autoconf to enforce libxenstore is present?

Reply:
Yes it would, and HAVE_XENSTORE is set by autoconf.
rpc-broker gives remote access to a dbus instance, not necessarily in a virtualized environment.
Building it and running it in OpenXT allows for v4v connections through the interposer, but it builds and runs just as well on bare-metal Linux, which allows for things like travis-ci testing.

Reply:
That is legitimate case, so disregard.